image/make-prep.sh

```diff
@@ -0,0 +1,325 @@
+# or the .zip) and it will create a new image file to write. You then
+# boot that image on a Raspberry Pi 2, and run the prepare-base.sh script
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.*
+
+SCRIPT_NAME=$(basename $0)
+VERBOSE=0
+ENABLE_CONSOLE=1
+ENABLE_WIFI=0
+WIFI_SSID=
+WIFI_PASSWORD=
+ENABLE_SSH=0
+HOSTNAME=gateway
+
+###########################################################################
+#
+# Prints the program usage
+#
+usage() {
+  cat <<END
+${SCRIPT_NAME} [OPTION] img-file prep-file
+
+where OPTION can be one of:
+
+  --noconsole     Don't enable the serial console
+  --ssh           Enable ssh
+  --ssid          Specify the SSID for Wifi access
+  --password      Specify the password for wifi access
+  --hostname      Specify the hostname
+  -h, --help      Print this help
+  -v, --verbose   Turn on some verbose reporting
+END
+}
+
+###########################################################################
+#
+# Copies the prepare-base.sh and prepare-base-root.sh files to the image
+#
+copy_prep_files() {
+  echo "Copying prepare-base scripts"
+
+  HOME_PI="${ROOT_MOUNTPOINT}/home/pi"
+
+  sudo rm -f "${HOME_PI}"/prepare-base*.sh
+  sudo wget --directory-prefix="${HOME_PI}" "https://raw.githubusercontent.com/mozilla-iot/gateway/master/image/prepare-base.sh"
+  sudo wget --directory-prefix="${HOME_PI}" "https://raw.githubusercontent.com/mozilla-iot/gateway/master/image/prepare-base-root.sh"
+  sudo chmod +x "${HOME_PI}"/prepare-base*.sh
+}
+
+###########################################################################
+#
+# Enables the serial console
+#
+enable_serial_console() {
+
+  CONFIG=${BOOT_MOUNTPOINT}/config.txt
+  if grep -q "enable_uart=1" ${CONFIG} ; then
+    echo "Enabling serial console"
+    sudo sh -c "cat >> '${CONFIG}'" <<END
+
+enable_uart=1
+force_turbo=1
+core_freq=250
+END
+  else
+    echo "Serial console appears to already be enabled"
+  fi
+}
+
+###########################################################################
+#
+# Enables SSH
+#
+enable_ssh() {
+  echo "Enabling SSH"
+  sudo touch ${BOOT_MOUNTPOINT}/ssh
+}
+
+###########################################################################
+#
+# Enables Wifi
+#
+enable_wifi() {
+  echo "Enabling Wifi"
+  WPA_CONF=${ROOT_MOUNTPOINT}/wpa_supplicant/wpa_supplicant.conf
+  sudo sh -c "cat >> '${CONFIG}'" <<END
+
+network={
+  ssid="${WIFI_SSID}"
+  password="${WIFI_PASSWORD}"
+}
+END
+}
+
+###########################################################################
+#
+# Update hostname
+#
+update_hostname() {
+  HOSTS_FILENAME="${ROOT_MOUNTPOINT}/etc/hosts"
+  HOSTNAME_FILENAME="${ROOT_MOUNTPOINT}/etc/hostname"
+  sudo sh -c "echo '${HOSTNAME}' > '${HOSTNAME_FILENAME}'"
+  sudo sed -i "${HOSTS_FILENAME}" -e "s/raspberrypi/${HOSTNAME}/"
```

Comment:
Might want to add the `g` flag to the regex, in case you have something like `127.0.0.1 raspberrypi raspberrypi.local`.